WordPress/gutenberg
# Mobile - RCTAztecView - Fixes font customization not getting updated

**Related PRs:**
- https://github.com/wordpress-mobile/gutenberg-mobile/pull/6036
- https://github.com/wordpress-mobile/WordPress-iOS/pull/21269

## What?
This PR fixes an issue that started to happen after the recent React Native upgrade where the post title and some other functionality related to custom font size and font styles like bold formatting wouldn't work as expected.

## Why?
It appears after setting the props `fontSize`, `fontWeight` wouldn't reflect the changes when rendering or typing new content. This is an iOS-only issue and it was affecting the setup we currently have with Aztec.

## How?
By forcing to update the current `font`,  `defaultFont`, and the placeholder's font as well as the `typingAttributes` to keep them updated after the font props are set.

I couldn't find another way to address this issue than by doing this proposal. We can explore other options if needed.

## Testing Instructions

## Scenario 1 - Post title placeholder shows the correct font styling

- Open the app using the build linked in this PR
- Start a new post
- **Expect** to see the post title's placeholder with the bold font styling and expected font family
- Type a title
- **Expect** the title to have the bold font styling and the expected font family

## Scenario 2 - Test the text formatting options

- Open the app using the build linked in this PR
- Start a new post
- Write some text content
- Test using the bold, italic, link, and text color formatting options
- **Expect** these options to work as expected
- Save the post
- Open it again and **expect** to see the formats previously set

## Scenario 3 - Font size and line height

- Open the app using the build linked in this PR
- Start a new post
- Write some text content
- Change the font size and line height for the Paragraph blocks (a block-based theme is needed)
- **Expect** to see the selected values
- Save the post
- Open the post and **expect** to see the previously selected values

### Testing Instructions for Keyboard
N/A

## Screenshots or screencast 

Expected behaviour
-|
<video src="https://github.com/WordPress/gutenberg/assets/4885740/587fdb89-fffb-435b-baba-1429b565f0a2" />|
